spec/support/saml_response_helper.rb

```diff
@@ -18,10 +25,27 @@ def input_id
       'SAMLResponse'
     end
 
-    def response_doc
-      return Nokogiri::XML(xml_response) if @test_type == 'feature'
+    def raw_xml_response
+      if @test_type == 'feature'
+        xml_response
+      else
+        @response.body
+      end
+    end
 
-      Nokogiri::XML(@response.body)
+    def response_doc
+      if raw_xml_response =~ /EncryptedData/
+        @original_encrypted = true
+        Nokogiri::XML(
+          OneLogin::RubySaml::Response.new(
+            raw_xml_response,
+            settings: sp1_saml_settings
+          ).decrypted_document.to_s
```

Comment:
I'm worried about the `to_s` here.  If this is returning a nokogiri document, then this is going to pretty print i think which inserts tabs into the content area which will scrambling the signature.

I thin kyou need to use `to_xml` or `serialiize` (not at a place where I can look it up) with a save option that disables adding spaces.